BedrockServer.cpp

```diff
@@ -2106,52 +2102,101 @@ void BedrockServer::_finishPeerCommand(unique_ptr<BedrockCommand>& command) {
+
+        // Lock _portMutex so suppressing the port does not cause it to be null
+        // in the middle of this function.
+        lock_guard<mutex> lock(_portMutex);
 
-    // Lock _portMutex so suppressing the port does not cause it to be null
-    // in the middle of this function.
-    lock_guard<mutex> lock(_portMutex);
+        for (auto& p : _portPluginMap) {
+            portList.push_back(reference_wrapper<const unique_ptr<Port>>(p.first));
+        }
 
-    for (auto& p : _portPluginMap) {
-        portList.push_back(reference_wrapper<const unique_ptr<Port>>(p.first));
-    }
+        // Try each port.
+        for (auto portWrapper : portList) {
+            const unique_ptr<Port>& port = portWrapper.get();
 
-    // Try each port.
-    for (auto portWrapper : portList) {
-        const unique_ptr<Port>& port = portWrapper.get();
+            // Skip null ports (if the command or control port are closed).
+            if (!port) {
+                continue;
+            }
 
-        // Skip null ports (if the command or control port are closed).
-        if (!port) {
-            continue;
-        }
+            // Accept as many sockets as we can.
+            while (true) {
+                sockaddr_in addr;
+                int s = S_accept(port->s, addr, true); // Note that this sets the newly accepted socket to be blocking.
 
-        // Accept as many sockets as we can.
-        while (true) {
-            sockaddr_in addr;
-            int s = S_accept(port->s, addr, true); // Note that this sets the newly accepted socket to be blocking.
+                // If we got an error or no socket, done accepting for now.
+                if (s <= 0) {
+                    break;
+                }
 
-            // If we got an error or no socket, done accepting for now.
-            if (s <= 0) {
-                break;
-            }
+                // Otherwise create the object for this new socket.
+                SDEBUG("Accepting socket from '" << addr << "' on port '" << port->host << "'");
+                Socket socket(s, Socket::CONNECTED);
+                socket.addr = addr;
 
-            // Otherwise create the object for this new socket.
-            SDEBUG("Accepting socket from '" << addr << "' on port '" << port->host << "'");
-            Socket socket(s, Socket::CONNECTED);
-            socket.addr = addr;
+                // If it came from a plugin, record that.
+                auto plugin = _portPluginMap.find(port);
+                if (plugin != _portPluginMap.end()) {
+                    socket.data = plugin->second;
+                }
 
-            // If it came from a plugin, record that.
-            auto plugin = _portPluginMap.find(port);
-            if (plugin != _portPluginMap.end()) {
-                socket.data = plugin->second;
-            }
+                // And start up this socket's thread.
+                _outstandingSocketThreads++;
```

Comment:
Should we move this into the `try` block where we define set `t = thread(...)`? Or maybe in between the `while(!threadStarted)` and the `try`? 

probably not the end of the world but as is I think it's possible this number gets off slightly in the case where we end up throwing a system_error since we should have incremented it but then if we failed to create the thread it means `handleSocket` never gets run which would decrement it, I think? 

also NAB but the comment above this I think pertains more to the `while` logic so probably should be just above that though I'm not sure it's a super useful comment either so maybe just remove it.

Reply:
We will either catch `system_error` and eventually retry and succeed, or we'll log an error and crash anyway, so I think this is fine as-is.